sdk/search/Azure.Search.Documents/tests/Samples/FieldBuilder/FieldBuilder.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Search.Documents.Indexes.Models;
+#if EXPERIMENTAL_SPATIAL
+using Azure.Core.Spatial;
+#endif
+
+namespace Azure.Search.Documents.Tests.Samples
+{
+    /// <summary>
+    /// Builds field definitions for a search index by reflecting over a user-defined model type.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This <see cref="FieldBuilder"/> was ported from the Microsoft.Azure.Search.Service package
+    /// to make migrating from using Microsoft.Azure.Search to Azure.Search.Documents easier.
+    /// It also uses System.Text.Json instead of Newtonsoft.Json (JSON.NET).
+    /// </para>
+    /// <para>
+    /// This is only a sample you can include in your code and future implementations may change
+    /// to follow modern guidelines and design principles.
+    /// </para>
+    /// </remarks>
+    public static class FieldBuilder
+    {
+        private static readonly IReadOnlyDictionary<Type, SearchFieldDataType> s_primitiveTypeMap =
+            new ReadOnlyDictionary<Type, SearchFieldDataType>(
+                new Dictionary<Type, SearchFieldDataType>()
+                {
+                    [typeof(string)] = SearchFieldDataType.String,
+                    [typeof(int)] = SearchFieldDataType.Int32,
+                    [typeof(long)] = SearchFieldDataType.Int64,
+                    [typeof(double)] = SearchFieldDataType.Double,
+                    [typeof(bool)] = SearchFieldDataType.Boolean,
+                    [typeof(DateTime)] = SearchFieldDataType.DateTimeOffset,
+                    [typeof(DateTimeOffset)] = SearchFieldDataType.DateTimeOffset,
+#if EXPERIMENTAL_SPATIAL
+                    [typeof(PointGeometry)] = SearchFieldDataType.GeographyPoint,
+#endif
+                });
+
+        private static readonly ISet<Type> s_unsupportedTypes =
+            new HashSet<Type>
+            {
+                typeof(decimal),
+            };
+
+        private static JsonNamingPolicy CamelCaseResolver { get; } = JsonNamingPolicy.CamelCase;
+
+        private static JsonNamingPolicy DefaultResolver { get; } = DefaultJsonNamingPolicy.Shared;
```

Comment:
I would remove these and let ObjectSerializer own this logic.

Reply:
This is a straight-up port of track 1 and the final API will look/work much differently.